frontend/apps/remark42/app/common/cookies.ts

```diff
@@ -27,15 +26,66 @@ export function setCookie(name: string, value: string, options: CookieOptions =
   let updatedCookie = `${name}=${value}`;
 
   for (const [key, value] of Object.entries(options)) {
-    updatedCookie += `; ${key}`;
-    if (value !== true) {
-      updatedCookie += `=${value}`;
+    // For boolean attributes like 'secure', only add them if true, otherwise skip
+    if (typeof value === 'boolean') {
+      if (value) {
+        updatedCookie += `; ${key}`;
+      }
+      // Skip false boolean values entirely
+    } else if (value !== undefined) {
+      // For non-boolean values
+      updatedCookie += `; ${key}=${value}`;
     }
```

Comment:
```suggestion
    if (value === true) {
        updatedCookie += `; ${key}`;
      // Skip false boolean values entirely
    }
    if (typeof value !== 'boolean') {
      // For non-boolean values
      updatedCookie += `; ${key}=${value}`;
    }
```

Reply:
I've applied it but I am almost sure it doesn't work the same as it's not accounting for undefined.